lib/usagereporter/teleport/aggregating/reporter.go

```diff
@@ -61,6 +61,8 @@ type ReporterConfig struct {
 	// HostID is the host ID of the current Teleport instance, added to reports
 	// for auditing purposes. Required.
 	HostID string
+	// AnonymizationKey is the key used to anonymize data user or resource names. Optional.
+	AnonymizationKey string
```

Comment:
Since the anonymization key is stored separately we could stop carrying around the whole `types.ClusterName` around and have just a string with the name itself in `ClusterName`, and that'd ensure that we don't use the cluster ID anywhere anymore. WDYT?

The same goes for the streaming reporter.

Reply:
ClusterName (anonymized with anonkey) is passed into reporter below along with HostID:
https://github.com/gravitational/teleport/blob/eac9ab21c058d519e5991b133ed24ade1a3b0dee/lib/usagereporter/teleport/aggregating/reporter.go#L118-L119
And then used in usage activity reports:
https://github.com/gravitational/teleport/blob/eac9ab21c058d519e5991b133ed24ade1a3b0dee/lib/usagereporter/teleport/aggregating/service.go#L54-L62

Reply:
I meant that we could be configuring the reporter with just the cluster name (the string currently returned by `cfg.ClusterName.GetClusterName()`) instead of the `types.ClusterName` item (`cfg.ClusterName`, including both cluster name and cluster ID), now that the anonymization key is given as a separate string.

Reply:
It is a bit nicer that way, but would require a multi-step process not to break `e`, so I'm thinking it's not worth it at this point.